lib/stack_master/sparkle_formation/compile_time/value_validator.rb

```diff
@@ -0,0 +1,40 @@
+module StackMaster
+  module SparkleFormation
+    module CompileTime
+      class ValueValidator
+
+        attr_reader :is_valid, :error
+
+        def validate
+          @is_valid = check_is_valid
+          @error = create_error unless @is_valid
+        end
+
+        protected
+
+        def check_is_valid
+          true
```

Comment:
The pattern for ensuring a sub-class implements this is to just raise the `NotImplemented` exception

Reply:
^ @shivaman